### PR TITLE
use standardized getStats api

### PR DESCRIPTION
**Pull request by yiming-amzn**
# standardized getStats api migration
* Enforced the usage of the standardized getStats API in the browser.
* Included audio level support for the standardized getStats API.
* Made minor adjustments to the values to ensure alignment with the results of the legacy getStats API.
* Removed the isRemote attribute from the getStats logic, as it was initially introduced by Firefox but later removed in 2019. More details can be found in this blog post: [Link to Mozilla's blog post on getStats isRemote](https://blog.mozilla.org/webrtc/getstats-isremote-66/).

### Test
* Tested in Chrome, Chrome Beta, Firefox, and Edge to ensure proper population of all values.
* Developed a custom ccp solution that collects both legacy getStats and standardized getStats API for Chrome, allowing us to verify the alignment of the values. Please see test result attached
* [legacyStats.txt](https://github.com/aws/connect-rtc-js/files/11826677/legacyStats.txt)
[newStas.txt](https://github.com/aws/connect-rtc-js/files/11826678/newStas.txt)

### CODE REVIEW

File: src/js/rtp-stats.js [hunk 1]
```diff
@@ -62,42 +61,32 @@ export function extractMediaStatsFromStats(timestamp, stats, streamType) {
 
                 }
             // Standardized report for input stream stats
-            } else if (statsReport.type === 'inbound-rtp' && !statsReport.isRemote && (streamType === 'audio_input' || streamType === 'video_input')) {
```
Comment (Vignendra) [inline src/js/rtp-stats.js:65]: Do we need the above legacy stats extraction logic if we are moving to standardized stats?
We support only the latest 3 browser versions so we should be good to remove lines 17 - 62?

File: src/js/rtp-stats.js [hunk 2]
```diff
@@ -62,42 +61,32 @@ export function extractMediaStatsFromStats(timestamp, stats, streamType) {
...
                 reportType = statsReport.type;
-                extractedStats = {
-                    timestamp:          timestamp,
-                    packetsLost:        statsReport.packetsLost,
-                    packetsCount:       packetsSent,
-                    rttMilliseconds:    Number.isInteger(statsReport.roundTripTime) ? statsReport.roundTripTime : is_defined(statsReport.roundTripTime) ? statsReport.roundTripTime * 1000 : null,
-                    jbMilliseconds:     when_defined(statsReport.jitter, 0) * 1000
-                };
+                extractedStats = extractedStats || {};
+                extractedStats.timestamp=timestamp;
```
Comment (Vignendra) [inline src/js/rtp-stats.js:88]: Nit: Space after and before =
Applies to below 3 lines and rest of the places

File: src/js/rtp-stats.js (hunk 1 above)
Comment (yiming-amzn) [reply src/js/rtp-stats.js:65]: Yes, you are right. I was initially considering enabling the standardized path first, then deprecating and cleaning up  the legacy code path as a follow up task, but it doesn't seem to provide any benefits. In the next revision, I will simply remove them. Thank you

File: src/js/rtp-stats.js (hunk 2 above)
Comment (yiming-amzn) [reply src/js/rtp-stats.js:88]: will address in the next revision, thanks!

File: src/js/rtp-stats.js [hunk 3]
```diff
@@ -8,96 +8,38 @@ import { is_defined, when_defined } from './utils';
...
-            } else if (statsReport.type === 'remote-inbound-rtp' && (streamType === 'audio_output' || streamType === 'video_output')) {
+            } else if (statsReport.type === 'outbound-rtp' && streamType === 'audio_output') {
+                // outbound-rtp: Stats for stream from CCP to Voice, as seen on the browser
+                extractedStats = extractedStats || {};
+                // packetsCount: number of packet sent by CCP, as seen on the browser
+                extractedStats.packetsCount = statsReport.packetsSent;
+            } else if (statsReport.type === 'media-source' && streamType === 'audio_output') {
+                extractedStats = extractedStats || {};
+                // Multiplying audioLevel by 32768 aligns its value with the legacy getStats API.
+                extractedStats.audioLevel = is_defined(statsReport.audioLevel) ? Math.floor(statsReport.audioLevel * 32768) : null;
```
Comment (siwei-w) [inline src/js/rtp-stats.js:34]: nit: can we make 32768 as a constant variable?

**Review by siwei-w — APPROVED** [review]

File: src/js/rtp-stats.js [hunk 4]
```diff
@@ -60,27 +60,29 @@ export function extractMediaStatsFromStats(timestamp, stats, streamType) {
                     };
 
                 }
-            // Standardized report for input stream stats
             } else if (statsReport.type === 'inbound-rtp' && streamType === 'audio_input') {
+                // inbound-rtp: Stats for stream from Voice to CCP, as seen on the browser
```
Comment (Vignendra) [inline src/js/rtp-stats.js:15]: Nit: Don't use `Voice` . Abstract away internal service details.

**Review by Vignendra — APPROVED** [review]

**Pull request merged**